README documentation refactoring (#476)
instana/go-sensor: docs/manual_instrumentation.md

## Instrumenting Code Manually

> The IBM Instana Go Tracer is built on top of the Opentracing SDK.

Reviewer: ```suggestion
The IBM Instana Go Tracer is built on top of the [Opentracing SDK](https://github.com/opentracing/opentracing-go).
```

In practical terms this means that we provide concrete implementations for Opentracing's tracer, span interfaces and other required implementations to fulfill the SDK.

All these concrete implementations are publicly available and can be used by anyone.

The main difference between customers creating their own traces and using our SDK is that we encapsulate all the boilerplate code and data collection logic within the tracer and additional packages instrumentation for your convenience.

> However, customers could want to create custom spans or provide extra tags with our SDK.

Reviewer: ```suggestion
However, customers can extend it to create custom spans or provide extra tags with our SDK.
```

### Creating and Reporting Spans

You can create an entry span with the `StartSpan` method and by providing `ext.SpanKindRPCServer` as one of the options:

```go
package main

import (
  instana "github.com/instana/go-sensor"
  ot "github.com/opentracing/opentracing-go"
  "github.com/opentracing/opentracing-go/ext"
)

func main() {
  col := instana.InitCollector(&instana.Options{
    Service: "My Service",
  })

  ps := col.StartSpan("my-entry-span", []ot.StartSpanOption{
    ext.SpanKindRPCServer,
  }...)

  // Do some work

  // Always make sure to call Finish to send the span to the Agent.
  ps.Finish()
}
```

The `StartSpan` method is compliant to the Opentracing interfaces, so you can rely on the Opentracing elements, such as `ot.StartSpanOption` for the span options or `ot.Tags` as part of the span options.
You can also notice the usage of Opentracing's `ext.SpanKindRPCServer` to define the span as an entry span.

> [!NOTE]
> You can use Opentracing's `ext.SpanKindRPCClient` to define an exit span. If no kind is provided, the span is assumed to be an `intermediate` span.

### Correlating Spans

> If you want to have a combination between spans, that is, multiple spans correlated to each other to form a more elaborate trace, you can use the method `ot.ChildOf()` as one of the options for child spans.

Reviewer: ```suggestion
If you wish to define a relation for a span with respect to another span, so that multiple spans can be correlated to each other to form a more elaborate trace, you can use the method `ot.ChildOf()` as one of the options for child spans.
```
